### PR TITLE
🌱  Move generic provider interface to API package

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

This PR moves generic provider interface into API package. This allows to drop wrapper objects from internal package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: api/v1alpha2/infrastructureprovider_wrapper.go
Comment (alexander-demicev) [inline api/v1alpha2/infrastructureprovider_wrapper.go:1]: should we name these files `*provider_wrapper.go?

File: controllers/alias.go
Comment (alexander-demicev) [inline controllers/alias.go:1]: I'm ok with dropping this alias, not sure if anyone is using a controller like this

**Review by alexander-demicev — APPROVED** [review]

**Comment by Fedosin** [comment]
Hmm... I tried to do something similar, but @JoelSpeed wanted to keep api package as simple as possible.  I'll let him leave a comment.

**Comment by Danil-Grigorev** [comment]
This change does not bring additional API dependencies from the currently used set.

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**